### PR TITLE
Add CmiAbortHook wrapper around CmiAbort for charm4py

### PR DESCRIPTION
**Pull request by juanjgalvez**

### CODE REVIEW

File: src/ck-core/ck.C [hunk 1]
```diff
@@ -2203,6 +2203,7 @@ void registerArrayMapProcNumExtCallback(int (*cb)(int, int, const int *)) {
 
 int CkMyPeHook() { return CkMyPe(); }
 int CkNumPesHook() { return CkNumPes(); }
+void CmiAbortHook(const char *msg) { CmiAbort(msg); }
```
Comment (matthiasdiener) [inline src/ck-core/ck.C]: Might be better to do CmiAbort(“%s”, msg)

Comment (evan-charmworks) [reply src/ck-core/ck.C]: I agree

**Comment by matthiasdiener** [comment]
Can you comment briefly why this patch is necessary for charm4py?

**Comment by juanjgalvez** [comment]
> Can you comment briefly why this patch is necessary for charm4py?

I don't write C code directly in Charm4py. I support 3 different interfaces to access the shared library, and they all expect to find a C function in libcharm.so with signature `CmiAbort(const char*)`. I don't think function attributes are supported in any of them. If there is some support for it, I have no idea how it works.

**Comment by juanjgalvez** [comment]
> Can you comment briefly why this patch is necessary for charm4py?
> 
> I don't write C code directly in Charm4py. I support 3 different interfaces to access the shared library, and they all expect to find a C function in libcharm.so with signature CmiAbort(const char*). I don't think function attributes are supported in any of them. If there is some support for it, I have no idea how it works.

Also, aren't function attributes a compile time thing? In some cases, like ctypes, there isn't even any compilation.

**Comment by matthiasdiener** [comment]
> > Can you comment briefly why this patch is necessary for charm4py?
> 
> I don't write C code directly in Charm4py. I support 3 different interfaces to access the shared library, and they all expect to find a C function in libcharm.so with signature `CmiAbort(const char*)`. I don't think function attributes are supported in any of them. If there is some support for it, I have no idea how it works.

But can’t you change the signature that the interfaces expect to CmiAbort(const char *fmt, ...) ?

**Comment by juanjgalvez** [comment]
> Can you comment briefly why this patch is necessary for charm4py?
> 
> I don't write C code directly in Charm4py. I support 3 different interfaces to access the shared library, and they all expect to find a C function in libcharm.so with signature CmiAbort(const char*). I don't think function attributes are supported in any of them. If there is some support for it, I have no idea how it works.
> 
> But can’t you change the signature that the interfaces expect to CmiAbort(const char *fmt, ...) ?

Possibly. I'll need some time to test this, since there are 3 different interfaces and they work differently. Will have to test on Windows too. If it works I guess we can just revert these patches.

**Pull request merged**